lock-impl/src/test/java/com/palantir/lock/impl/LegacyTimelockServiceIntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    @AfterEach
+    public void afterEach() {
+        lockService.close();
+    }
+
+    @Test
+    public void canAcquireAndReleaseMultipleLeases() {
+        TestLease lease1 = acquire(TIMESTAMP_LEASE_NAME1, TIMESTAMP_LEASE_NAME2);
+        TestLease lease2 = acquire(TIMESTAMP_LEASE_NAME1);
+        TestLease lease3 = acquire(TIMESTAMP_LEASE_NAME2);
+        TestLease lease4 = acquire(TIMESTAMP_LEASE_NAME1, TIMESTAMP_LEASE_NAME2);
+        assertThat(lease1.minLeasedTimestamp())
+                .isEqualTo(lease2.minLeasedTimestamp())
+                .isEqualTo(lease3.minLeasedTimestamp())
+                .isEqualTo(lease4.minLeasedTimestamp());
+        assertThatMinLeaseTimestamps(TIMESTAMP_LEASE_NAME1, TIMESTAMP_LEASE_NAME2)
+                .containsEntry(TIMESTAMP_LEASE_NAME1, lease1.minLeasedTimestamp())
+                .containsEntry(TIMESTAMP_LEASE_NAME2, lease1.minLeasedTimestamp());
+
+        // Release middle leases
+        lease2.unlock();
+        assertThatMinLeaseTimestamps(TIMESTAMP_LEASE_NAME1, TIMESTAMP_LEASE_NAME2)
+                .containsEntry(TIMESTAMP_LEASE_NAME1, lease1.minLeasedTimestamp())
+                .containsEntry(TIMESTAMP_LEASE_NAME2, lease1.minLeasedTimestamp());
+        lease3.unlock();
+        assertThatMinLeaseTimestamps(TIMESTAMP_LEASE_NAME1, TIMESTAMP_LEASE_NAME2)
+                .containsEntry(TIMESTAMP_LEASE_NAME1, lease1.minLeasedTimestamp())
+                .containsEntry(TIMESTAMP_LEASE_NAME2, lease1.minLeasedTimestamp());
+
+        // Release oldest lease
+        lease1.unlock();
+
+        // TODO(jakubk): This is awkward because we don't have the leased timestamps in the response.
+        MapAssert<TimestampLeaseName, Long> timestampsAssert =
+                assertThatMinLeaseTimestamps(TIMESTAMP_LEASE_NAME1, TIMESTAMP_LEASE_NAME2);
+        timestampsAssert.extractingByKey(TIMESTAMP_LEASE_NAME1).satisfies(minLeaseTimestamp -> assertThat(
+                        minLeaseTimestamp)
+                .isGreaterThan(lease4.minLeasedTimestamp()));
+        timestampsAssert.extractingByKey(TIMESTAMP_LEASE_NAME2).satisfies(minLeaseTimestamp -> assertThat(
+                        minLeaseTimestamp)
+                .isGreaterThan(lease4.minLeasedTimestamp()));
+    }
+
+    @ReviewedRestrictedApiUsage
+    private TestLease acquire(TimestampLeaseName... leases) {
+        Map<TimestampLeaseName, Integer> request = StreamEx.of(leases)
+                .mapToEntry(_ignored -> ThreadLocalRandom.current().nextInt(1, 10))
+                .toMap();
+        TimestampLeaseResults timestampLeaseResults = timelock.acquireTimestampLeases(request);
+        return createTestLease(request, timestampLeaseResults);
+    }
+
+    private MapAssert<TimestampLeaseName, Long> assertThatMinLeaseTimestamps(TimestampLeaseName... expectedKeys) {
+        return assertThat(getMinLeasedTimestamps(Set.of(expectedKeys))).containsOnlyKeys(Set.of(expectedKeys));
+    }
+
+    @ReviewedRestrictedApiUsage
```

Comment:
We should not need these in tests, I forgot to add
```
allowedOnPath = ".*/src/test/.*", // Unsafe behavior in tests is ok.
```

Where the restricting annotation is used.

Reply:
Ok, happy for you to FLUP with this.